gematria/proto/execution_annotation.proto

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package gematria;
+
+// Contains a register index (the LLVM register number) and the value that
+// the register should contain when setting up the block for benchmarking.
+message RegisterAndValue {
+  // The LLVM register number that refers to the register in question.
+  uint32 register_index = 1;
+
+  // The value that the register should be filled with.
+  int64 register_value = 2;
+}
+
+// Contains the necessary annotations needed to execute a basic block.
+message ExecutionAnnotations {
+  // The location in memory where the benchmark code should be loaded in.
+  uint64 code_location = 1;
+
+  // The size of the memory blocks that should be allocated by the execution
+  // runtime prior to benchmarking.
+  uint64 block_size = 2;
+
+  // The contents that memory blocks should be filled with.
+  uint64 block_contents = 3;
+
+  // Addresses where memory should be allocated by the execution runtime.
+  repeated uint64 accessed_blocks = 4;
```

Comment:
Maybe in the future. That isn't supported by either of the current annotators though, and doesn't work well with the the current benchmarking strategy of mapping everything to a single physical page so that everything easily fits in the L1D cache.

I could see this making sense for future extensions, but there's no need for it currently, there's nothing needing it on the road map, and thus to me it seems like it would be making something too generic for little purpose.